src/components/ShopifyCDNImage.ts

```diff
@@ -0,0 +1,72 @@
+import { defineComponent } from "vue"
+import { resourceUrl, defaultImgUrl } from "../index";
+
+export default defineComponent({
+  template: `
+            <img :src="imageUrl"/>
+            `,
+  data() {
+    return {
+      imageUrl: ''
+    }
+  },
+  props: ['src', 'size'],
+  async mounted() {
+    this.setImageUrl();
+  },
+  updated() {
+    this.setImageUrl();
+  },
+  methods: {
+    addSizeInImgUrl(src: string, size: string) {
+      if (src) {
+        // return original size if no size is given
+        if (!size) return src
+        return src
+        // remove any current image size then add the new image size
+        .replace(/_(pico|icon|thumb|small|compact|medium|large|grande|original|1024x1024|2048x2048|master)+\./g, '.')
+        .replace(/\.jpg|\.png|\.gif|\.jpeg/g, function (match) {
+          return '_' + size + match;
+        })
+      }
+    },
+    checkIfImageExists(src: string) {
+      return new Promise((resolve, reject) => {
+        const img = new Image();
+        img.onload = function () {
+          resolve(true);
+        }
+        img.onerror = function () {
+          reject(false);
+        }
+        img.src = src;
+      })
+    },
+    setImageUrl() {
+      const src = this.addSizeInImgUrl(this.src, this.size)
```

Comment:
For assets or images from the resource server, this is not needed. As the component is specific to Shopify, we could remove the handling for assets as well resource server for now.

Reply:
Removed support for resourceUrl